tuna/parse_args.py

```diff
@@ -164,3 +165,67 @@ def args_check(args: argparse.Namespace, parser: argparse.Namespace) -> None:
     parser.error(
         "When setting up a new tunning session the following must be specified: "\
         "label.")
+
+
+@typing.no_type_check
+def get_import_cfg_parser(
```

Comment:
@cderb 

mypy complains about this function when placed in the same file:

`
**mypy tuna/parse_args.py --ignore-missing-imports --follow-imports=skip**
tuna/parse_args.py:179: error: Item "None" of "Optional[Namespace]" has no attribute "add_mutually_exclusive_group"
tuna/parse_args.py:186: error: Item "None" of "Optional[Namespace]" has no attribute "add_argument"
tuna/parse_args.py:191: error: Item "None" of "Optional[Namespace]" has no attribute "add_argument"
tuna/parse_args.py:197: error: Item "None" of "Optional[Namespace]" has no attribute "add_argument"
tuna/parse_args.py:202: error: Item "None" of "Optional[Namespace]" has no attribute "add_argument"
tuna/parse_args.py:208: error: Item "None" of "Optional[Namespace]" has no attribute "add_argument"
tuna/parse_args.py:213: error: Item "None" of "Optional[Namespace]" has no attribute "add_argument"
tuna/parse_args.py:220: error: Item "None" of "Optional[Namespace]" has no attribute "add_argument"
Found 8 errors in 1 file (checked 1 source file)
`

Reply:
Changed setup arg parser intakes args from this:


```
def setup_arg_parser(desc: str,
                     arg_list: List[TunaArgs],
                     parser: argparse.Namespace = None,
                     with_yaml: bool = True) -> Optional[argparse.Namespace]:
```

                     
to this 

```
def setup_arg_parser(desc: str,
                     arg_list: List[TunaArgs],
                     parser: Optional[jsonargparse.ArgumentParser] = None,
                     with_yaml: bool = True) -> jsonargparse.ArgumentParser:
```

Tried casting to None, then Pylint complained about branches - likely requiring the setup_arg_parser function to be split into part.

Reply:
removing the Optional from the returned value would definitely require more work, lets go with what @cderb said.